mitmproxy/addons/xss.py

```diff
@@ -0,0 +1,391 @@
+from mitmproxy import ctx  # Used for logging information to mitmproxy
+from socket import gaierror, gethostbyname  # Used to check whether a domain name resolves
+from urllib.parse import urlparse  # Used to modify paths and queries for URLs
+import requests  # Used to send additional requests when looking for XSSs
+import re  # Used for pulling out the payload
+from html.parser import HTMLParser  # used for parsing HTML
```

Comment:
I generally appreciate code with lots of comments, but you can leave imports uncommented. If I'd be interested to see what `gethostbyname` is used for, I'd grep the file for `gethostbyname` and hopefully understand its use at the respective place in the code. :)